packages/client/src/components/interface/Header/Header.tsx

```diff
@@ -254,69 +258,100 @@ class HeaderComp extends React.Component<IFullProps, IState> {
   }
 
   getMobileHeaderActionProps(activeMenuItem: ACTIVE_MENU_ITEM, theme: ITheme) {
+    const locationId = new URLSearchParams(this.props.location.search).get(
+      'locationId'
+    )
     if (activeMenuItem === ACTIVE_MENU_ITEM.PERFORMANCE) {
       return {
-        mobileLeft: {
-          icon: () => this.hamburger(),
-          handler: this.toggleMenu
-        },
-        mobileRight: {
-          icon: () => <Activity stroke={theme.colors.primary} />,
-          handler: () =>
-            this.props.mapPerformanceClickHandler &&
-            this.props.mapPerformanceClickHandler()
-        }
-      }
-    } else if (
-      this.props.userDetails?.role &&
-      USERS_WITHOUT_SEARCH.includes(this.props.userDetails?.role)
-    ) {
-      return {
-        mobileLeft: {
-          icon: () => this.hamburger(),
-          handler: this.toggleMenu
-        }
-      }
-    } else if (activeMenuItem === ACTIVE_MENU_ITEM.USERS) {
-      if (this.props.mapPinClickHandler) {
-        return {
-          mobileLeft: {
+        mobileLeft: [
+          {
             icon: () => this.hamburger(),
             handler: this.toggleMenu
-          },
-          mobileRight: {
-            icon: () => <Location inverse />,
+          }
+        ],
+        mobileRight: [
+          {
+            icon: () => <Activity stroke={theme.colors.primary} />,
             handler: () =>
-              this.props.mapPinClickHandler && this.props.mapPinClickHandler()
+              this.props.mapPerformanceClickHandler &&
+              this.props.mapPerformanceClickHandler()
           }
+        ]
+      }
+    } else if (activeMenuItem === ACTIVE_MENU_ITEM.USERS) {
+      if (this.props.mapPinClickHandler) {
+        return {
+          mobileLeft: [
+            {
+              icon: () => this.hamburger(),
+              handler: this.toggleMenu
+            }
+          ],
+          mobileRight: [
+            {
+              icon: () => <SearchBlue />,
+              handler: () =>
+                this.props.mapPinClickHandler && this.props.mapPinClickHandler()
```

Comment:
I think we should change the name of the function `mapPinClickHandler`. It is so confusing name.